e2e/libs/node/operations/aws.py

```diff
@@ -0,0 +1,103 @@
+from node.operations.abstract_cloud_provider import AbstractCloudProvider
+from node import Nodes
+
+import boto3
+import logging
+
+class EC2(AbstractCloudProvider):
+
+    _ec2_instance = boto3.resource('ec2')
+
+    def __init__(self) -> None:
+        super().__init__()
+
+    def get_all_node_instances(self):
+        result = []
+        for node in Nodes.all_nodes:
+            instances = self.get_node_instance(node['name'])
+            result.append(instances)
+        return result
+
+    def get_node_instance(self, node_name):
+        if node_name == "":
+            raise Exception(f"empty parameter: node_name={node_name}")
+
+        logging.info(f"getting node {node_name} EC2 instances")
+
+        for node in Nodes.all_nodes:
+            if node['name'] == node_name:
+                ip_address = node['ip_address']
+        if ip_address == "":
+            raise Exception(f"{node_name}'s ip address is empty")
+
+        filter = [{'Name': 'ip-address','Values': [ip_address]}]
+        instances = self._ec2_instance.instances.filter(Filters=filter)
+        if len(list(instances)) > 0:
+            return instances
+
+        filter = [{'Name': 'private-ip-address','Values': [ip_address]}]
+        instances = self._ec2_instance.instances.filter(Filters=filter)
+        return instances
+
+
+    def reboot_node_instance(self, node_name):
+        if node_name == "":
+            raise Exception(f"empty parameter: node_name={node_name}")
+
+        node_instances = self.get_node_instance(node_name)
+        if len(node_instances) == 0:
+            logging.warn(f"cannot find node instance")
+            return
+
+        for instance in node_instances:
+            instance.reboot()
+            logging.debug(f'EC2 instance has beend rebooted:', {node_name})
+            instance.wait_until_running()
+            logging.debug(f'EC2 instance "{node_name}" has been running')
+        logging.info("fiished rebooting node instances")
```

Comment:
```suggestion
        logging.info("finished rebooting node instances")
```

And please fix others.